src/helpers.rs

```diff
@@ -305,3 +306,25 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         self.eval_libc(name)?.to_i32()
     }
 }
+
+
+pub fn bytes_to_os_string<'tcx>(bytes: Vec<u8>) -> InterpResult<'tcx, OsString> {
```

Comment:
IMO the better APIs would be `read_os_string` and `write_os_string`, also taking care of the interaction with memory.

Reply:
I started doing it that way but then i realized that there are several cases where writing is conditional. For example in `getcwd`, the current directory is only written if it fits into the buffer. Then by symmetry I decided to do the same with reading.

Reply:
But this way you are still leaving a bunch of duplicate code in the various handlers.

The `write_os_string` anyway needs a buffer size as argument, *all* write functions do. `strcpy` was a mistake in C. ;)

Reply:
Oh great :P I'll change it